recipes/libbacktrace/all/conanfile.py

```diff
@@ -83,8 +74,8 @@ def generate(self):
         env.generate()
 
         tc = AutotoolsToolchain(self)
-        if (self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version) >= "12") or \
-           (self.settings.compiler == "msvc" and Version(self.settings.compiler.version) >= "180"):
+        if is_msvc(self):
+            # https://github.com/conan-io/conan/issues/6514
```

Comment:
what's the advantage of such modification? We have this pattern in all autotools based recipes, it allows to at least give a change to consumer to build with old Visual Studio versions, while this change prevents it.

Reply:
libbacktrace doesnt build with visual studio 2012 - this follows the pattern of raising the exception in validate(), in the same way it already does when trying to build the shared library when using Visual Studio, and many other recipes do in similar circumstances.

By the time it reaches this block, `validate()` has already run. In the unlikely event that someone tries to revisit vs2012 support for this library, I believe the comment and the PR history provide enough context to address this.

Reply:
It doesn't build with Visual Studio 2012 due to libbacktrace source code or as a consequence of https://github.com/conan-io/conan/issues/12739?

Reply:
Due to the former, as I mention in the pull request description.

Reply:
I've read the description but I was not sure it was not coming from this issue a side effect. I don't disagree with this modification, I just want to mention that users have already submitted issues in conan-center due to -FS injected in Visual Studio 2012 so it should be removed carefully in other recipes if there is a wish to purge this pattern in other recipes.